cli/src/commands/git/clone.rs

```diff
@@ -57,6 +57,9 @@ pub struct GitCloneArgs {
     /// Whether or not to colocate the Jujutsu repo with the git repo
     #[arg(long)]
     colocate: bool,
+    /// Create a shallow clone of the given depth.
+    #[arg(long)]
+    depth: Option<usize>,
```

Comment:
Nit: We don't end help message sentences with a period. And this could be 
```suggestion
   #[arg(long, default = 0)]
   depth: usize,
```